documentation/Using-the-SDK/Creating-a-simple-sdk-plugin.md

```diff
@@ -1,75 +1,98 @@
-# Creating a Simple Custom Data Source
+# Creating a Simple SDK Plugin
 
-A Custom Data Source (CDS) uses the SDK to create a data processing pipeline for any data sources, typically files, 
-that the CDS supports. Notably, a CDS acts as the entry point for the SDK runtime to discover and utilizes these data 
-pipelines. An SDK Plugin may contain more than one CDS.
+An SDK plugin uses the SDK to create a mapping between arbitrary data sources (e.g. files with 
+specific formats) and 
+* zero or more __tables__
+* zero or more __data outputs__
 
-A standard CDS will utilize a Custom Data Processor (CDP) to process the data sources that the SDK runtime provides it.
 
-A plugin may also define tables that can be used by viewers such as Windows Performance Analyzer (WPA) to display the 
+A `CustomDatasource` (CDS) acts as the entry point for the SDK runtime to discover and create these mappings. 
+An SDK plugin may contain more than one CDS, but it is __highly recommended__ to only include one CDS per 
+assembly inside a plugin.
+
+A standard CDS will utilize a `CustomDataProcessor` (CDP) to process the data sources that the SDK runtime provides it.
+
+A plugin may also define __tables__ that can be used by viewers such as Windows Performance Analyzer (WPA) to display the 
 processed data in an organized, interactable collection of rows. Tables are discovered and passed into CDPs by the SDK 
 runtime.
 
 The SDK also supports advanced data processing pipelines through data cookers and extensions, but these topics are 
 covered in the advanced documentaton. Refer to [Overview](../Overview.md) for more information.
 
-## Corresponding Sample
+In this tutorial, we will explore creating a simple SDK plugin that has one CDS, one CDP, and one table. 
+Refer to [the following sample](../../samples/SimpleDataSource/SampleAddIn.csproj) for source code that implements the steps outlined in this file.
 
-Refer to [the following sample Data Source](../../samples/SimpleDataSource/SampleAddIn.csproj) for source code that implements the steps outlined in this file.
+## Requirements of a Simple Plugin
 
-## Requirements of a Simple Data Source
-
-The following are required for a Simple Data Source:
+To create a simple plugin, you must:
 
 1. Create a public class that implements the abstract class `CustomDataSourceBase`. This is your Custom Data Source 
-   (CDS).
+   (CDS)
 2. Create a public class that implements the abstract class `CustomDataProcessorBase`. This your Custom Data Processor 
-   (CDP).
-3. Create one or more data table classes. These classes must:
-   - Be public and static.
-   - Be decorated with `TableAttribute`.
-   - Expose a static public field or property named TableDescriptor of type `TableDescriptor` which provides information
+   (CDP)
+3. Create or more data table classes. These classes must:
+   - Be public
```

Comment:
should this be public and static still? and if not why did this change?

Reply:
They are not required to be static. It's recommended, but not mandatory.